lib/index.coffee

```diff
@@ -1,57 +1,196 @@
+        'C:\\Python34',
+        'C:\\Python3.5',
+        'C:\\Python35',
+        'C:\\Program Files (x86)\\Python 2.7',
+        'C:\\Program Files (x86)\\Python 3.4',
+        'C:\\Program Files (x86)\\Python 3.5',
+        'C:\\Program Files (x64)\\Python 2.7',
+        'C:\\Program Files (x64)\\Python 3.4',
+        'C:\\Program Files (x64)\\Python 3.5',
+        'C:\\Program Files\\Python 2.7',
+        'C:\\Program Files\\Python 3.4',
+        'C:\\Program Files\\Python 3.5'
+      ]
+      path_env = (env.Path or '')
+    else
+      paths = ['/usr/local/bin', '/usr/bin', '/bin', '/usr/sbin', '/sbin']
+      path_env = (env.PATH or '')
+
+    path_env = path_env.split(path.delimiter)
+    path_env.unshift(pythonPath if pythonPath and pythonPath not in path_env)
+    for p in paths
+      if p not in path_env
+        path_env.push(p)
+    env.PATH = path_env.join(path.delimiter)
+
     pi = new AtomIsort()
+    pi.python_env = env
+
 
     {CompositeDisposable} = require 'atom'
     @subs = new CompositeDisposable
 
-    @subs.add atom.commands.add 'atom-workspace', 'pane:active-item-changed', ->
-      pi.removeStatusbarItem()
+    if atom.config.get('atom-isort.showStatusBar')
+      @subs.add atom.commands.add 'atom-workspace', 'pane:active-item-changed', ->
+        pi.removeStatusbarItem()
 
-    @subs.add atom.commands.add 'atom-workspace', 'atom-isort:sortImports', ->
+    @subs.add atom.commands.add 'atom-text-editor[data-grammar="source python"]','atom-isort:sort imports', ->
       pi.sortImports()
 
-    @subs.add atom.commands.add 'atom-workspace', 'atom-isort:checkImports', ->
+    @subs.add atom.commands.add 'atom-text-editor[data-grammar="source python"]','atom-isort:check imports', ->
       pi.checkImports()
 
     @subs.add atom.config.observe 'atom-isort.sortOnSave', (value) ->
       atom.workspace.observeTextEditors (editor) ->
         if value
-          editor._isortSort = editor.onDidSave -> pi.sortImports()
+          editor._isortSort = editor.buffer.onWillSave -> pi.sortImports(editor, true)
         else
           editor._isortSort?.dispose()
 
     @subs.add atom.config.observe 'atom-isort.checkOnSave', (value) ->
       atom.workspace.observeTextEditors (editor) ->
         if value
-          editor._isortCheck = editor.onDidSave -> pi.checkImports()
+          editor._isortCheck = editor.buffer.onWillSave -> pi.checkImports(editor, true)
         else
           editor._isortCheck?.dispose()
 
-    StatusDialog = require './status-dialog'
-    @status = new StatusDialog pi
-    pi.setStatusDialog(@status)
+    @subs.add atom.config.observe 'atom-isort.showStatusBar', (value) ->
+      atom.workspace.observeTextEditors (editor) ->
+        if not value
+          if pi.statusDialog?
+            pi.removeStatusbarItem()
+        else
+          # TODO: re-adding status bar isn't working, not sure how to get it to.
+          # Scope in JS is confusing. Need to call consumeStatusBar?
```

Comment:
Yeah this was originally written before the status bar was a consumable service. It doesn't surprise me that things don't quite work right when re-adding the status bar.